python/python/bystro/utils/config.py

```diff
@@ -0,0 +1,41 @@
+"""Utility functions for accessing project-level configuration files."""
+from pathlib import Path
+from typing import Dict, Union
+
+from ruamel.yaml import YAML
+
+# A ConfigDict has string keys and values of type ConfigDictValue, or
+# a list of ConfigDictValues.  A ConfigDictValue is a str, int, float,
+# or (nested) ConfigDict.  ConfigDictValue can be extended with
+# additional primitive types if necessary.
+
+
+ConfigDict = Dict[str, Union["ConfigDictValue", list["ConfigDictValue"]]]
+ConfigDictValue = Union[str, int, float, "ConfigDict"]
+
+
+def get_bystro_project_root() -> Path:
+    """Return Path of bystro project root."""
+    # find project root by walking up the tree until we get to top level bystro directory.
+    # The bystro top level is assumed to be the unique directory containing a startup.yml file.
+    path = Path().absolute()
+    FILESYSTEM_ROOT = Path("/")
+    found_startup_file = False
+    while path != FILESYSTEM_ROOT:
+        if any(path.glob("startup.yml")):
+            found_startup_file = True
```

Comment:
Could we memoize this so that we're not hitting the filesystem a bunch if we rely on this, lifting BYSTRO_ROOT to a config-level global?

Reply:
good call, thanks